src/mds/MDCache.cc

```diff
@@ -14109,7 +14102,7 @@ void MDCache::upkeep_main(void)
         if (active_with_clients) {
           trim_client_leases();
         }
-        if (is_ready_to_trim_cache() || mds->is_standby_replay()) {
+        if (is_open() || mds->is_standby_replay()) {
```

Comment:
Under what condition will we reach this function and not be open?

Reply:
This was added by @batrick in https://github.com/ceph/ceph/pull/48483. I went through the code and got that only when the mds is in `up:active` state will the `open == true`.